sklearn/preprocessing/data.py

```diff
@@ -1767,135 +1755,185 @@ class OneHotEncoder(BaseEstimator, TransformerMixin):
     sklearn.feature_extraction.FeatureHasher : performs an approximate one-hot
       encoding of dictionary items or strings.
     """
-    def __init__(self, n_values="auto", categorical_features="all",
-                 dtype=np.float64, sparse=True, handle_unknown='error'):
-        self.n_values = n_values
+
+    def __init__(self, classes='auto', categorical_features="all",
+                 n_values='auto', dtype=np.float64, sparse=True,
```

Comment:
default for `n_values` should be `None` or some other sentinel so that you can warn the user of deprecation if they've explicitly specified `'auto'`